src/core/buffers/buffer_orchestrator.ts

```diff
@@ -188,7 +189,13 @@ export default function BufferOrchestrator(
       areComplete ? EVENTS.endOfStream() : EVENTS.resumeStream()
     ));
 
-  return observableMerge(...buffersArray,
+  // TODO do not reload whole content when the decipherability changes
```

Comment:
I think a better solution might be to:
  - stop the concerned PeriodBuffers (discriminating on Adaptation type) 
  - clean the buffer based on infos returned by the SegmentBookkeeper
  - restart the concerned PeriodBuffers

Reply:
But that's more work